sdk/tools/spec2def/spec2def.c

```diff
@@ -711,52 +744,151 @@ OutputLine_def(FILE *fileDest, EXPORT *pexp)
+void
+Fatalv(
+    const char* filename,
+    unsigned nLine,
+    const char *pcLine,
+    const char *pc,
+    unsigned errorlen,
+    const char *format,
+    va_list argptr)
 {
-    char *pc, *pcLine;
-    int nLine;
+    unsigned i, errorpos, len;
+
+    /* Get the length of the line */
+    for (len = 0; pcLine[len] != '\x0D' && pcLine[len] != '\x0A'; len++)
+        (void)0;
+
+    if (pc == NULL)
+    {
+        pc = pcLine + len - 1;
+        errorlen = 1;
+    }
+
+    errorpos = pc - pcLine;
+
+    /* Output the error message */
+    fprintf(stderr, "ERROR: (%s:%u:%u): ", filename, nLine, errorpos);
+    vfprintf(stderr, format, argptr);
+    fprintf(stderr, "\n");
+
+    /* Output the line with the error */
+    fprintf(stderr, "> %.*s\n", len, pcLine);
+
+    if (errorlen == 0)
+    {
+        errorlen = TokenLength(pc);
+    }
+
+    for (i = 0; i < errorpos + 2; i++)
+    {
+        fprintf(stderr, " ");
+    }
+    for (i = 0; i < errorlen; i++)
+    {
+        fprintf(stderr, "~");
+    }
+    fprintf(stderr, "\n");
+    exit(-1);
+}
+
+void
+Fatal(
+    const char* filename,
+    unsigned nLine,
+    const char *pcLine,
+    const char *pc,
+    unsigned errorlen,
+    const char *format,
+    ...)
+{
+    va_list argptr;
+
+    va_start(argptr, format);
+    Fatalv(filename, nLine, pcLine, pc, errorlen, format, argptr);
+    va_end(argptr);
+}
+
+EXPORT *
+ParseFile(char* pcStart, FILE *fileDest, unsigned *cExports)
+{
+    EXPORT *pexports;
+    const char *pc, *pcLine;
+    int cLines, nLine;
     EXPORT exp;
-    int included, version_included;
-    char namebuffer[16];
+    int included;
     unsigned int i;
 
+    *cExports = 0;
+
     //fprintf(stderr, "info: line %d, pcStart:'%.30s'\n", nLine, pcStart);
 
+    /* Count the lines */
+    for (cLines = 1, pcLine = pcStart; *pcLine; pcLine = NextLine(pcLine), cLines++)
+    {
+        /* Nothing */
+    }
+
+    /* Allocate an array of EXPORT structures */
+    pexports = (EXPORT*)malloc(cLines * sizeof(EXPORT));
+    if (pexports == NULL)
+    {
+        fprintf(stderr, "ERROR: %s: failed to allocate EXPORT array of %u elements\n", pszSourceFileName, cLines);
+        return 0;
+    }
+
     /* Loop all lines */
     nLine = 1;
     exp.nNumber = 0;
     for (pcLine = pcStart; *pcLine; pcLine = NextLine(pcLine), nLine++)
     {
         pc = pcLine;
 
+        exp.strName.buf = NULL;
+        exp.strName.len = 0;
+        exp.strTarget.buf = NULL;
+        exp.strTarget.len = 0;
         exp.nArgCount = 0;
         exp.uFlags = 0;
         exp.nNumber++;
+        exp.nStartVersion = 0;
+        exp.nEndVersion = 0xFFFFFFFF;
+        exp.bVersionIncluded = 1;
 
         /* Skip white spaces */
         while (*pc == ' ' || *pc == '\t') pc++;
 
-        /* Skip empty lines, stop at EOF */
+        /* Skip empty lines */
         if (*pc == ';' || *pc <= '#') continue;
-        if (*pc == 0) return 0;
+
+        /* On EOF we are done */
+        if (*pc == 0)
```

Comment:
This is covered by `*pc <= '#'` above so seemingly can't be reached.

Reply:
Should be ```== '#'```